de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/editor/GapBuffer.java

```diff
@@ -0,0 +1,115 @@
+package de.fu_berlin.inf.dpp.server.editor;
+
+public class GapBuffer {
+    // Buffer
+    private char[] content;
+
+    // Gap
+    private int gapStart;
+    private int gapLength;
+
+    public GapBuffer(String content) {
+       this(content, 1);
```

Comment:
Is there a reason that the `GapBuffer` is initialized with a buffer size of 1 by default?
I know this is a very vague question, but 1 seems like a weird choice to me.
It reserves ram in anticipation of future insertions to save computation time, but still only allocates so little buffer space that the array size will have to be increased after one insertion. And if the first action is a deletion, this one additional character of buffer size won't make much of a difference.

But I have no experience with gap buffers, so maybe I am missing something (or am overthinking this).